include/miopen/miopen.h

```diff
@@ -4338,6 +4397,143 @@ MIOPEN_EXPORT miopenStatus_t miopenDropoutBackward(miopenHandle_t handle,
 /** @} */
 // CLOSEOUT DROPOUT DOXYGEN GROUP
 
+// TensorReduce APIs
+/** @addtogroup TensorReduce
+ *
+ *  @{
+ */
+
+/*! @brief Creates the ReduceTensor descriptor object
+ *
+ * @param reduceTensorDesc Pointer to a ReduceTensor descriptor type
+ * @return            miopenStatus_t
+ */
+MIOPEN_EXPORT miopenStatus_t
+miopenCreateReduceTensorDescriptor(miopenReduceTensorDescriptor_t* reduceTensorDesc);
+
+/*! @brief Destroy the ReduceTensor descriptor object
+ *
+ * @param reduceTensorDesc  ReduceTensor descriptor type (input)
+ * @return            miopenStatus_t
+ */
+MIOPEN_EXPORT miopenStatus_t 
+miopenDestroyReduceTensorDescriptor(miopenReduceTensorDescriptor_t reduceTensorDesc); 
+
+/*! @brief Initialize a ReduceTensor descriptor object
+ *
+ * @param reduceTensorDesc         Pointer to the ReduceTensor descriptor object (output)
+ * @param reduceTensorOp           Enumerant specifying the operation used by ReduceTensor (input)
+ * @param reduceTensorCompType     Enumerant specifying the data type used with ReduceTensor
+ * operation (input)
+ * @param reduceTensorNanOpt       Enumerant specifying the Nan number propagation mode (input)
+ * @param reduceTensorIndices      Enumerant specifying the indices modes used by ReduceTensor
+ * (input)
+ * @param reduceTensorIndicesType  Enumerant specifying the data type of the indices (input)
+ * @return           miopenStatus_t
+ */
+MIOPEN_EXPORT miopenStatus_t
+miopenSetReduceTensorDescriptor(miopenReduceTensorDescriptor_t reduceTensorDesc,
```

Comment:
There should be simpler call that just takes `reduceTensorOp` and `reduceTensorCompType`. And ideally nan propogation and indices should be set with additional set methods.